examples/tls-sni/envoy.yaml

```diff
@@ -0,0 +1,149 @@
+static_resources:
+  listeners:
+  - address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 10000
+    listener_filters:
+    - name: "envoy.filters.listener.tls_inspector"
```

Comment:
im unclear about this

in the sni section "securing envoy" page i did not mention it, so i  tested out here whether it was needed to make the sni work and it doesnt seem to be

reading the  docs again (https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/tls_inspector#tls-inspector) it seems like it does 2 things - make sni work, and add stats

it doesnt seem like it really is required to make sni work - it this correct ?

Reply:
IIRC TLS inspector is dynamically inserted if not specified when using SNI matching, so this is an explicit version of what the code does internally. @ggreenway @PiotrSikora is that correct?

Reply:
Yes, that's correct. I think it should be included in all examples that use SNI for filter chain matching.

Reply:
OK sounds good. Let's merge!

Reply:
> IIRC TLS inspector is dynamically inserted

> Yes, that's correct. I think it should be included in all examples

im not sure best how to reconcile these 2 statements - im thinking if its dynamically inserted its not strictly necessary

i guess i should add it in the quick-start page - and perhaps comment that its implied if not included

Reply:
I'm saying that anytime tls_inspector is needed, I think examples should explicitly add it to configuration, even if the config would work fine without it (due to it being automatically added). 

This makes it easier (in my opinion) to understand why it works, and also clearer what to do if you need to build a more complicated config.

For example, if the config needs both proxy protocol and tls inspector, it matters which order those filters are added. If they're explicit in the config, it's clear they're in the right order.

Reply:
I agree that it should be explicitly included. Also, Envoy prints an error when it auto-injects it.

Reply:
cool - so here its there in the config - ill add to quick start tomorrow